.github/workflows/static.yml

```diff
@@ -0,0 +1,34 @@
+name: Static and unit checks
+
+on: pull_request
+
+jobs:
+  static:
+    runs-on: ubuntu-latest
+    name: "Running ${{ matrix.test }} tests with perl v${{ matrix.perl-version }}"
+    strategy:
+      matrix:
+        include:
+          - test: static
+            perl-version: 5.26
+          - test: unit
+            perl-version: 5.26
```

Comment:
not sure if you extend this step by step and I assume you are aware but just to ensure nothing goes missed: So far we have three parallel jobs "static", "unit", "compile", here only two are specified so far.

Reply:
Yes, one of the things I would do straight away is to do compile check on changed files only, and full check once merged or even just once a day.
Also generation of perl doc is not added yet as will need more changes to make it work.

Reply:
Let me quote yourself from https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11545#discussion_r535345488

"Let me make it works as it was in travis before we start with improvements ;)"